src/app/media-viewer-wrapper/media-viewer-wrapper.component.ts

```diff
@@ -20,22 +20,31 @@ export class MediaViewerWrapperComponent implements OnInit {
   unsupportedType = 'txt';
 
   selectedTab = 'pdf';
+  url = this.pdfUrl;
+  annotationSet = pdfAnnotationSet;
+  comments = [];
+
   showToolbar = true;
   showAnnotations = false;
-
   toolbarButtons = new ToolbarButtonToggles();
   showCommentSummary = new Subject<boolean>();
-  comments = [];
-
-  imageAnnotationSet = imageAnnotationSet;
-  pdfAnnotationSet = pdfAnnotationSet;
 
   constructor(
     public api: AnnotationApiService
   ) {}
 
-  selectTab(currentTab: string) {
-    this.selectedTab = currentTab;
+  selectTab(newTab: string) {
+    this.selectedTab = newTab;
+
+    if (newTab === 'pdf') {
+      this.url = this.pdfUrl;
+      this.annotationSet = pdfAnnotationSet;
+    } else if (newTab === 'image') {
+      this.url = this.imageUrl;
+      this.annotationSet = imageAnnotationSet;
+    } else {
+      this.url = this.unsupportedUrl;
+    }
```

Comment:
maybe we could pass the url and annotationSet through the method to avoid the if-else

Reply:
swings and roundabouts, I'd probably keep it as is. How strongly do you feel about it?